[BUG] Python interperter distclean isn't working

### Description / Steps to reproduce the issue

Try to compile python interpreter using
```
$ cd nuttx
$ make distclean
$ ./tools/configure.sh rv-virt:python
$ make -j$(nproc)
$ ls -l nuttx
```

Run `make distclean or `make apps_distclean`
Python compile objects are still there.

Dry run doesn't show removal of the files

```
make apps_distclean -n
make -C /home/peter/nuttx/apps/interpreters/ distclean APPDIR="/home/peter/nuttx/apps"
rm -f  Kconfig
rm -f  .kconfig
```

### On which OS does this issue occur?

[OS: Linux]

### What is the version of your OS?

Ubuntu 22.04

### NuttX Version

master

### Issue Architecture

[Arch: risc-v]

### Issue Area

[Area: Interpreters]

### Verification

- [X] I have verified before submitting the report.

```
peter@SMW015329:~/nuttx/apps/interpreters/python$ git clean -ffdx
Removing Python/
Removing build/
Removing install/
Removing v3.13.0.zip
peter@SMW015329:~/nuttx/apps/interpreters/python$ ls
Kconfig  Make.defs  Makefile  Setup.local  config.site  mount_modules.c  patch
```

```
$ ./tools/configure.sh rv-virt:python
$ make -j$(nproc)

# Lots of output

checking size of fpos_t... 0
checking size of size_t... configure: error: in `/home/peter/nuttx/apps/interpreters/python/build/target':
configure: error: cannot compute sizeof (size_t)
See `config.log' for more details
make[2]: *** [Makefile:107: /home/peter/nuttx/apps/interpreters/python/build/target/Makefile] Error 77
make[1]: *** [Makefile:57: /home/peter/nuttx/apps/interpreters/python_depend] Error 2
make: *** [tools/Unix.mk:630: pass2dep] Error 2
peter@SMW015329:~/nuttx/nuttx$
peter@SMW015329:~/nuttx/nuttx$ make distclean
```

Compile failure I'll try `make distclean`

```
peter@SMW015329:~/nuttx/apps/interpreters/python$ ls
Kconfig  Make.defs  Makefile  Python  Setup.local  build  config.site  install  mount_modules.c  patch  v3.13.0.zip
```

So yeah most likely `make distclean` would work on a working build, but a build fail then you can't clean through `make distclean`

Also which compiler are you using? Because I think my failure has to do with my gcc version.

> ```
> peter@SMW015329:~/nuttx/apps/interpreters/python$ git clean -ffdx
> Removing Python/
> Removing build/
> Removing install/
> Removing v3.13.0.zip
> peter@SMW015329:~/nuttx/apps/interpreters/python$ ls
> Kconfig  Make.defs  Makefile  Setup.local  config.site  mount_modules.c  patch
> ```
> 
> ```
> $ ./tools/configure.sh rv-virt:python
> $ make -j$(nproc)
> 
> # Lots of output
> 
> checking size of fpos_t... 0
> checking size of size_t... configure: error: in `/home/peter/nuttx/apps/interpreters/python/build/target':
> configure: error: cannot compute sizeof (size_t)
> See `config.log' for more details
> make[2]: *** [Makefile:107: /home/peter/nuttx/apps/interpreters/python/build/target/Makefile] Error 77
> make[1]: *** [Makefile:57: /home/peter/nuttx/apps/interpreters/python_depend] Error 2
> make: *** [tools/Unix.mk:630: pass2dep] Error 2
> peter@SMW015329:~/nuttx/nuttx$
> peter@SMW015329:~/nuttx/nuttx$ make distclean
> ```
> 
> Compile failure I'll try `make distclean`
> 
> ```
> peter@SMW015329:~/nuttx/apps/interpreters/python$ ls
> Kconfig  Make.defs  Makefile  Python  Setup.local  build  config.site  install  mount_modules.c  patch  v3.13.0.zip
> ```
> 
> So yeah most likely `make distclean` would work on a working build, but a build fail then you can't clean through `make distclean`
> 
> Also which compiler are you using? Because I think my failure has to do with my gcc version.

I see... Yeah, would need to investigate why it fails to clean the files if the build fails too...

About the toolchain, we always recommend using the same CI uses: https://nuttx.apache.org/docs/latest/platforms/risc-v/qemu-rv/boards/rv-virt/index.html#risc-v-toolchain

--- a/Application.mk
+++ b/Application.mk
@@ -393,6 +393,7 @@ clean::
 	$(call CLEANAROBJS)
 	$(call CLEAN)
 distclean:: clean
+	$(call DELFILE, .kconfig)
 	$(call DELFILE, Make.dep)
 	$(call DELFILE, .depend)
 

--- a/audioutils/lame/Makefile
+++ b/audioutils/lame/Makefile
@@ -102,6 +102,6 @@ ifneq ($(PREFIX),)
 endif
 
 distclean::
-	$(Q)cd $(DST_PATH) && make distclean
+	$(Q)if [ -d $(DST_PATH) ]; then cd $(DST_PATH) && $(MAKE) distclean; fi
 
 include $(APPDIR)/Application.mk

--- a/crypto/wolfssl/Makefile
+++ b/crypto/wolfssl/Makefile
@@ -168,7 +168,7 @@ distclean::
 	$(call DELDIR, $(WOLFSSL_UNPACKNAME))
 	$(call DELDIR, $(WOLFSSL_EXAMPLESNAME))
 else
-distclean:
+distclean::
 	$(call DELDIR, $(WOLFSSL_UNPACKNAME))
 endif
 

--- a/netutils/libshvc/Makefile
+++ b/netutils/libshvc/Makefile
@@ -82,7 +82,7 @@ ifeq ($(wildcard $(SHVDIR)/.git),)
 $(eval $(call RUN_OMK_MAKE, context, default-config include-pass))
 
 distclean::
-	make -C $(SHVDIR) distclean
+	if [ -d $(SHVDIR) ]; then make -C $(SHVDIR) distclean; fi
 	$(call DELDIR, $(SHVDIR))
 	$(call DELFILE, $(SHV_LIBS4C_TARGZ))
 endif

--- a/netutils/libulut/Makefile
+++ b/netutils/libulut/Makefile
@@ -82,7 +82,7 @@ ifeq ($(wildcard $(ULUTDIR)/.git),)
 $(eval $(call RUN_OMK_MAKE, context, default-config include-pass))
 
 distclean::
-	make -C $(ULUTDIR) distclean
+	if [ -d $(ULUTDIR) ]; then make -C $(ULUTDIR) distclean; fi
 	$(call DELDIR, $(ULUTDIR))
 	$(call DELFILE, $(ULUT_TARGZ))
 endif

